c_glib/arrow-glib/array-builder.cpp

```diff
@@ -65,13 +65,37 @@ garrow_array_builder_append_values(GArrowArrayBuilder *builder,
                   is_valids_length);
       return FALSE;
     }
-    uint8_t valid_bytes[is_valids_length];
-    for (gint64 i = 0; i < is_valids_length; ++i) {
-      valid_bytes[i] = is_valids[i];
+
+    const gint64 chunk_size = 4096;
+    gint64 n_chunks = is_valids_length / chunk_size;
+    gint64 n_remains = is_valids_length % chunk_size;
+    for (gint64 i = 0; i < n_chunks; ++i) {
+      uint8_t valid_bytes[chunk_size];
+      gint64 offset = chunk_size * i;
+      const gboolean *chunked_is_valids = is_valids + offset;
+      for (gint64 j = 0; j < chunk_size; ++j) {
+        valid_bytes[j] = chunked_is_valids[j];
+      }
+      status = arrow_builder->AppendValues(values + offset,
```

Comment:
I made a JIRA to extend the C++ interface to also support bytemaps as a reminder to make this easier: https://issues.apache.org/jira/browse/ARROW-2487

Reply:
Thanks.
I'll comment on it about this use case.